mne/preprocessing/ecg.py

```diff
@@ -219,6 +210,14 @@ def find_ecg_events(raw, event_id=999, ch_name=None, tstart=0.0,
                               thresh_value=qrs_threshold, l_freq=None,
                               h_freq=None, verbose=False)
 
+    if ecg_events.size == 0:
+        logger.info('Could not detect any ECG events.')
+        out = (ecg_events, idx_ecg, np.nan)
+        if return_ecg:
+            ecg = ecg[np.newaxis]
+            out += (ecg,)
+        return out
```

Comment:
Why does the code below not work nicely already? If we can make it support the 0-events case easily / with one or two conditionals it's better than short-circuiting, as it simplifies the logic and flow

Reply:
Pushed a commit to revert this